libp2p/protocols/kademlia/kademlia.nim

```diff
@@ -200,11 +317,14 @@ proc new*(
       # TODO: figure out why this fails:
       # error "could not handle request",
       #   peerId = conn.PeerId, err = getCurrentExceptionMsg()
-    finally:
-      await conn.close()
-
   return kad
 
+proc setSelector*(kad: KadDHT, selector: EntrySelector) =
+  kad.entrySelector = selector
+
+proc setValidator*(kad: KadDHT, validator: EntryValidator) =
+  kad.entryValidator = validator
+
```

Comment:
```suggestion
```
✂️ 
I don't like dumb getters and setters (dumb == ones that do no extra work for you)

Reply:
me neither, but a) there isn't a usecase I can anticipate that calls for a get, and removing the setter means that the field needs to be public, which means that it can be set to nil, thus causing a regression on the "if it's valid code, it's correct code" objective.

Reply:
Can you not do `kad.setValidator(nil)`?